src/base/bittorrent/session.cpp

```diff
@@ -4764,10 +4762,10 @@ void Session::handleSessionStatsAlert(const lt::session_stats_alert *p)
     m_status.diskWriteQueue = stats[m_metricIndices.peer.numPeersDownDisk];
     m_status.peersCount = stats[m_metricIndices.peer.numPeersConnected];
 
-    const int numBlocksRead = stats[m_metricIndices.disk.numBlocksRead];
-    const int numBlocksCacheHits = stats[m_metricIndices.disk.numBlocksCacheHits];
+    const auto numBlocksRead = stats[m_metricIndices.disk.numBlocksRead];
+    const auto numBlocksCacheHits = stats[m_metricIndices.disk.numBlocksCacheHits];
```

Comment:
Please keep the type visible (not everyone is using an IDE).

Reply:
I changed it to be more consistent with the stuff at lines 4720-4729. Or should those be changed to int as well, for consistency?

Reply:
>Or should those be changed to int as well, for consistency?

I would prefer this.

Reply:
IIRC using `auto` was because libtorrent uses different type for these numbers in libt 1.2. You should check that first.

Reply:
libtorrent is using `std::int64_t` so we should be ok. What about using qint64 or std::int64_t as well?

Reply:
>IIRC using auto was because libtorrent uses different type for these numbers since libt 1.2. You may want to check that first.

https://github.com/arvidn/libtorrent/blob/RC_1_1/include/libtorrent/alert_types.hpp#L1948
https://github.com/arvidn/libtorrent/blob/RC_1_2/include/libtorrent/alert_types.hpp#L2160

Reply:
> I think you should keep it `auto` until we drop support for libt 1.1.

Ok, thanks for pointing that out.

On a similar note, in such cases, what types should we prefer, and why, between `long`, `std::int64_t` and `qint64`? The latter two have the advantage of guaranteeing 64 bits, which could matter if someone is using some platform/compiler that does not guarantee that. But then what are the criteria for deciding to use Qt types over standard types, for such basic types? I noticed in the codebase does not use std::int64_t at all, and I'd like to know more about why that is.

Reply:
Just stick to the type where it comes from or use the type that minimizes type conversion (I would like to think std::int64_t is different than qint64 even if the underlying could be the same), in other cases you are free to choose as long as the choice is sensible.

>The latter two have the advantage of guaranteeing 64 bits

You only use them when you strictly require the variable to be exact 64 bit, otherwise you probably shouldn't.

Reply:
Solved as per https://github.com/qbittorrent/qBittorrent/pull/12086#discussion_r386000154